src/PortingAssistant.Compatibility.Core/Checkers/ExternalCompatibilityChecker.cs

```diff
@@ -190,9 +190,9 @@ public class PackageFromS3
         }
 
 
-        public async Task<PackageDetails> GetPackageDetailFromS3(string fileToDownload, IHttpService httpService, HashSet<string> apis = null)
+        public async Task<PackageDetails> GetPackageDetailFromS3(string fileToDownload, IRegionalDatastoreService regionalDatastoreService, HashSet<string> apis = null)

```

Comment:
IRegionalDatastoreService is a global variable, you don't need to add it to GetPackageDetailFromS3 parameter.   remove regionalDatastoreService from the method parameter.